contracts/RoundManager.sol

```diff
@@ -0,0 +1,193 @@
+ */
+contract RoundManager {
+
+    /* ============ Datatypes ============ */
+    enum Status {
+        Open,
+        Completed,
+        Cancelled
+    }
+
+    struct Round {
+        Status status;
+        uint256 startTime;
+        uint256 totalVotes;
+    }
+
+    /* ============ State Variables ============ */
+    // Stores rounds
+    Round[] public rounds;
+    // The registry of a specific DAO
+    ContributorRegistry public registry;
+    // List of addresses with admin privileges for a given DAO
+    address[] public admins;
+    // List of the addresses in the current round
+    mapping(uint256 => address[]) public usersInRound;
+    // Number of votes for a user in a given round
+    mapping(uint256 => mapping(address => uint256)) public numVotes;
+    // Mapping of round to those who have voted in that round
+    mapping(uint256 => mapping(address => bool)) public hasVoted; // Or just map uint256 to address and iterate through?
+    // TEMPORARY MAPPING
+    mapping(address => uint256) weightedVoting; // Use weighted voting identifier as way to communicate w registry about who can/can't vote
+    // Timeout
+    uint256 timePerRound;
+    // If the DAOs choose to consider time
+    bool timed;
+
+    /* ============ Events ============ */
+
+    event RoundOpened(uint256 roundID);
+    event RoundOpenFailed();
+    event RoundClosed(uint256 roundID, address closedBy);
+    event VoteCast(uint256 roundID, address voter);
+
+    /* Modifiers */
+
+    modifier canVote(uint256 roundID) {
+        require(!hasVoted[roundID][msg.sender] && weightedVoting[msg.sender] > 0);
+        _;
+    }
+
+    modifier roundState(uint256 roundID, Status _status) {
+        require(rounds[roundID].status == _status, "Round is not open");
+        _;
+    }
+    
+    modifier isAdmin() {
+        bool ret = false;
+        for(uint i = 0; !ret && i < admins.length; i++) {
+            if(admins[i] == msg.sender) {
+                ret = true;
+            }
+        }
+        require(ret);
+        _;
+    }
+
+    /* ============ Constructor ============ */
+
+    constructor(address registryAddress, uint256 _timePerRound, bool _timed) {
+        registry = ContributorRegistry(registryAddress);
+        admins.push(msg.sender);
+        timePerRound = _timePerRound;
+        timed = _timed;
+    }
+
+    /* ============ Mutating Functions ============ */
+
+    /**
+     * Opens a new round.
+     * Confirms that there exist valid users passed to the function before opening a new round.
+     * A new round opens with a given salary to be split  amongst the contributors following the voting period.
+     * 
+     *
+     * @param newUsers          Addresses of the proposed users to be eligible for votes
+     *
+     */
+    function openRound(
+        address[] memory newUsers
+    ) public isAdmin {
+        // Check to see if only round open
+        require(rounds.length == 0 || rounds[rounds.length - 1].status != Status.Open);
```

Comment:
I would add an informative message to this require statement.
Also I think we could add a requirment here for the "newUsers" array to be of length > 0 so we can fail early and save gas in ths scenario.